spaceship/lib/spaceship/client.rb

```diff
@@ -417,11 +417,32 @@ def login(user = nil, password = nil)
     # This method is used for both the Apple Dev Portal and iTunes Connect
     # This will also handle 2 step verification
     def send_shared_login_request(user, password)
-      # First we see if we have a stored cookie for 2 step enabled accounts
-      # this is needed as it stores the information on if this computer is a
-      # trusted one. In general I think spaceship clients should be trusted
-      load_session_from_file
+      # Check if we have a cached/valid session here
+      # Fixes
+      #   - https://github.com/fastlane/fastlane/issues/10812
+      #   - https://github.com/fastlane/fastlane/issues/10793
+      #
+      # Before 4th December 2017 we didn't load existing session from the disk
+      # but changed it, because Apple introduced a rate limit, which is fine by itself
+      # but unfortunately it also rate limits successful logins, meaning if you call multiple
+      # tools in a lane (e.g. call match 5 times), this would mean it locks you out of the account
+      # for a while.
+      # By loading existing sessions and checking if they're valid, we're sending less login requests
+      # More context on why this change was necessary https://github.com/fastlane/fastlane/pull/11108
+      #
+      if load_session_from_file
+        # Check if the session is still valid here
+        begin
+          # We use the olympus session to determine if the old session is still valid
+          # As this will raise an exception if the old session has expired
+          # If the old session is still valid, we don't have to do anything else in this method
+          return true if fetch_olympus_session.count > 0
+        rescue
```

Comment:
Just want to be sure that we arent swallowing an important error here? 🚀

Reply:
What error are you thinking about? If this raises an exception, we're gonna do a fresh login, and with that call `fetch_olympus_session` again

Reply:
Thanks for clarifying that @KrauseFx, that makes sense. We don't need to worry about messaging a failure here since later on we'll be calling `fetch_olympus_session` again, if it fails that time, we'll catch it and display an error.